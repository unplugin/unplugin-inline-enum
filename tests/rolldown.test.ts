import path from 'node:path'
import { rolldownBuild } from '@sxzz/test-utils'
import Oxc from 'unplugin-oxc/rolldown'
import { expect, test } from 'vitest'
import UnpluginInlineEnum from '../src/rolldown'

test('rolldown', async () => {
  const { snapshot } = await rolldownBuild(
    path.resolve(__dirname, 'fixtures/main.ts'),
    [
      UnpluginInlineEnum({
        scanDir: path.resolve(__dirname, 'fixtures'),
        scanMode: 'fs',
      }),
      Oxc(),
    ],
    {},
    { minify: 'dce-only' },
  )
  expect(snapshot).toMatchSnapshot()
})
