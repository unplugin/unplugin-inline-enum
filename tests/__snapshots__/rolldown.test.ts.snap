// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rolldown 1`] = `
"// main.js
const values = [
	"foo",
	100,
	4
];
console.log("foo", 100, 4);
console.log(values);
"
`;
